File: server/src/routes/compare.ts
import { Router } from 'express';
import { prisma } from '../shared/prisma.js';
import { z } from 'zod';

const router = Router();

router.get('/', async (req, res) => {
  const schema = z.object({ ids: z.string() });
  const parsed = schema.safeParse(req.query);
  if (!parsed.success) {
    return res.status(400).json({ error: 'ids query param required, e.g. ids=1,2,3' });
  }
  const ids = parsed.data.ids
    .split(',')
    .map((s) => s.trim())
    .filter(Boolean);

  const items = await prisma.course.findMany({ where: { course_id: { in: ids } } });
  res.json({ items });
});

export default router;

File: backend/routes/compare.py
from flask import Blueprint, request
from models.course import Course

compare_bp = Blueprint('compare', __name__)

@compare_bp.get('/')
def compare():
    ids = request.args.get('ids')
    if not ids:
        return {'error': 'ids query param required e.g. ids=CS101,CS102'}, 400
    
    id_list = [s.strip() for s in ids.split(',') if s.strip()]
    items = Course.query.filter(Course.course_id.in_(id_list)).all()
    return {'items': [c.to_dict() for c in items]}